docs/src/snap/howto/storage/cloud.md

```diff
@@ -0,0 +1,476 @@
+    namespace: kube-system
+```
+
+After a moment, you should see the cloud controller manager pod was
+successfully deployed.
+
+```bash
+NAME                                 READY   STATUS    RESTARTS        AGE
+aws-cloud-controller-manager-ndbtq   1/1     Running   1 (3h51m ago)   9h
+```
+
+## Deploy the EBS CSI Driver
+
+Now that the cloud controller manager is deployed and can communicate with AWS,
+you are ready to deploy the EBS CSI driver. The easiest way to deploy the
+driver is with the Helm chart. Luckily, Canonical K8s has a built-in helm
+command.
+
+If you want to create encrypted drives, you need to add the statement to the
+policy you are using for the instance.
+
+```json
+{
+  "Effect": "Allow",
+  "Action": [
+      "kms:Decrypt",
+      "kms:GenerateDataKeyWithoutPlaintext",
+      "kms:CreateGrant"
+  ],
+  "Resource": "*"
+}
+```
+
+Then, add the helm repo for the EBS CSI Driver.
+
+```bash
+sudo k8s helm repo add aws-ebs-csi-driver https://kubernetes-sigs.github.io/aws-ebs-csi-driver
+sudo k8s helm repo update
+```
+
+Finally, install the Helm chart, making sure to set the correct region as an
+argument.
+
+```bash
+sudo k8s helm upgrade --install aws-ebs-csi-driver \
+    --namespace kube-system \
+    aws-ebs-csi-driver/aws-ebs-csi-driver \
+    --set controller.region=us-east-2
+```
+
+Once the command completes, you can verify the pods are successfully deployed:
+
+```bash
+kubectl get pods -n kube-system -l app.kubernetes.io/name=aws-ebs-csi-driver
+```
+
+```bash
+NAME                                  READY   STATUS    RESTARTS        AGE
+ebs-csi-controller-78bcd46cf8-5zk8q   5/5     Running   2 (3h48m ago)   8h
+ebs-csi-controller-78bcd46cf8-g7l5h   5/5     Running   1 (3h48m ago)   8h
+ebs-csi-node-nx6rg                    3/3     Running   0               9h
+```
+
+The status of all pods should be "Running".
+
+## Deploy a workload
+
+Everything is in place for you to deploy a workload that dynamically creates
+and uses an EBS volume.
+
+First, create a StorageClass and a PersistentVolumeClaim:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: ebs-sc
+provisioner: ebs.csi.aws.com
+volumeBindingMode: WaitForFirstConsumer
+parameters:
+  type: gp3 # EBS volume type (gp3, gp2, etc.)
+  fsType: ext4 # File system type
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: ebs-pvc
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+  storageClassName: ebs-sc
+```
+
+Then, you can deploy a pod that uses a volume. Because we used
+`WaitForFirstConsumer`, you'll only see the volume in AWS once the pod is
+deployed.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app-using-ebs
+spec:
+  containers:
+  - name: app
+    image: nginx
+    volumeMounts:
+    - mountPath: "/data"
+      name: ebs-volume
+  volumes:
+  - name: ebs-volume
+    persistentVolumeClaim:
+      claimName: ebs-pvc
+```
```

Comment:
Maybe we can also show commands to create the StorageClass, PVC and the Pod.

Reply:
Not sure what you mean. Those objects are created by applying the YAML.

Reply:
I think Hue was saying to include the apply yaml command?